File: java-itba/1-3guias/introduccion/ComplexNumber.java
class ComplexNumber {

  private double real;
  private double imaginary;

  public ComplexNumber(double real, double imaginary) {
    this.real = real;
    this.imaginary = imaginary;
  }

  public ComplexNumber(double real) {
    this.real = real;
  }

  public ComplexNumber(ComplexNumber n1, ComplexNumber n2) {
    this.real = n1.get_real() + n2.get_real();
    this.imaginary = n1.get_imaginary() + n2.get_imaginary();
  }

  public double get_real() {
    return real;
  }

  public double get_imaginary() {
    return imaginary;
  }

}

File: ruby-itba/tp8_solucion/complex_number/ComplexNumber.rb
# La clase debe permitir que dos instancias de números complejos puedan sumarse retornado
# una nueva instancia con el resultado de la operación.
class ComplexNumber
  def initialize(real, imaginary)
    @real = real
    @imaginary = imaginary
  end

  def to_s
    "#{@real} + #{@imaginary}"
  end

  def +(other)
    return nil if other.nil? || !other.is_a?(ComplexNumber)

    ComplexNumber.new(@real + other.real, @imaginary + other.imaginary)
  end

  attr_reader :real, :imaginary
end
